### PR TITLE
restructuring and extending tests for maxLatency (tier2) update on patch endpoint

### PR DESCRIPTION
**Pull request by artHalMirantis**
also removed partial update for one of the tests (coverage was kept by other)

### CODE REVIEW

File: src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/PartialUpdateSpec.groovy [hunk 1]
```diff
@@ -85,10 +85,6 @@ class PartialUpdateSpec extends HealthCheckSpecification {
                         field   : "maxLatency",
                         newValue: 12345
                 ],
-                [
-                        field   : "maxLatencyTier2",
-                        newValue: 23456
-                ],
```
Comment (niksv) [inline src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/PartialUpdateSpec.groovy:91]: As you removed a positive test case - should we add  new one? When a flow has maxLatency and you are trying to PATCH maxLatencyTier2

Comment (artHalMirantis) [reply src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/PartialUpdateSpec.groovy:91]: test was failing due to new validation
i review other, and wanted to implement it with another fix
but you are right - so returning it back with right precondition to allow test run properly

File: src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/PartialUpdateSpec.groovy [hunk 2]
```diff
@@ -122,6 +126,10 @@ class PartialUpdateSpec extends HealthCheckSpecification {
         given: "A flow"
         def swPair = topologyHelper.switchPairs.first()
         def flow = flowHelperV2.randomFlow(swPair)
+        flow.tap{
+            pathComputationStrategy = "cost"
+            maxLatency = 1234
+        }
```
Comment (niksv) [inline src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/PartialUpdateSpec.groovy:132]: Why did you add it? Data was not changed. Maybe you wanted to add `maxLatencyTier2` into `data`?

File: src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/PartialUpdateSpec.groovy [hunk 3]
```diff
@@ -792,6 +800,39 @@ class PartialUpdateSpec extends HealthCheckSpecification {
...
+        errorDetails.errorMessage == "Invalid flow data"
+
+        cleanup:
+        flowHelperV2.deleteFlow(flow.flowId)
+
+        where:
+        maxLatencyBefore | maxLatencyT2Before | maxLatencyAfter | maxLatencyT2After
+        null             | null               | null            | 1
+        2                | 3                  | null            | 1
+        2                | null               | null            | 4
```
Comment (niksv) [inline src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/PartialUpdateSpec.groovy:833]: Why the last one is incorect? after update flow will has maxLatency 2, maxLatencyTier2 4. It's correct

Comment (artHalMirantis) [reply src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/PartialUpdateSpec.groovy]: was used as test parameter - missed for deleting
fixed (removed)

File: src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/PartialUpdateSpec.groovy (hunk 2 above)
Comment (artHalMirantis) [reply src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/PartialUpdateSpec.groovy]: removed (test with incorrect api version)

**Review by niksv — APPROVED** [review]

File: src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/PartialUpdateSpec.groovy [hunk 4]
```diff
@@ -792,6 +796,38 @@ class PartialUpdateSpec extends HealthCheckSpecification {
         defaultFlow && flowHelperV2.deleteFlow(defaultFlow.flowId)
     }
 
+    @Tidy
+    @Unroll("Unable to partial update flow (maxLatency #maxLatencyAfter and maxLatencyTier2 #maxLatencyT2After)")
+    def "Unable to partial update flow with maxLatency incorrect value"() {
+        given: "Two potential flows"
+        def flow = flowHelperV2.randomFlow(topologyHelper.switchPairs[0]).tap {
+            maxLatency = maxLatencyBefore
+            maxLatencyT2Before
```
Comment (rtretyak) [inline src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/PartialUpdateSpec.groovy:805]: `maxLatencyTier2 = maxLatencyT2Before` ?

Comment (artHalMirantis) [reply src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/PartialUpdateSpec.groovy]: right
its not crucial for the case, but it was missed
fixed now

**Review by pkazlenka — APPROVED** [review]
